webapp/app/assessment/fra/sections/originalDataPoint/actions/actions.js

```diff
@@ -120,12 +124,32 @@ export const copyPreviousNationalClasses = (countryIso, odp) => async (dispatch)
   }
 }
 
+export const updateNationalClassValue = (index, fieldName, valueCurrent, valueUpdate) => (dispatch, getState) => {
+  const state = getState()
+  const odp = OriginalDataPointState.getActive(state)
+  const countryIso = AppState.getCountryIso(state)
+  const odpUpdate = ODP.updateNationalClass(odp, index, fieldName, acceptNextDecimal(valueUpdate, valueCurrent))
+  dispatch(saveDraft(countryIso, odpUpdate))
+}
+
+// ====== Paste
+export const pasteNationalClassValues = (props) => (dispatch, getState) => {
+  const state = getState()
+  const odp = OriginalDataPointState.getActive(state)
+  const countryIso = AppState.getCountryIso(state)
+  const { evt, rowIndex, colIndex, columns, allowGrow = false, allowedClass = () => true } = props
+
+  const rawPastedData = readPasteClipboard(evt, 'string')
+  const { updatedOdp } = handlePaste(columns, allowedClass, odp, allowGrow, rawPastedData, rowIndex, colIndex)
```

Comment:
maybe it can be passed as `...props`? or grouped with "named params" (object as pasteNatioanlClassValues `props`)

Reply:
this action is used in several places. I should repeat that call several times, instead if it's here, only once